internal/provider/resource_vault_cluster.go

```diff
@@ -296,6 +339,10 @@ func resourceVaultClusterUpdate(ctx context.Context, d *schema.ResourceData, met
 	}
 
 	if d.HasChange("tier") {
+		if inPlusTier(string(cluster.Config.Tier)) {
```

Comment:
Follow-up suggestion: this could go in a [CustomizeDiff](https://www.terraform.io/plugin/sdkv2/resources/customizing-differences) function on the `tier` property above.